examples/5_secret_storage.py

```diff
@@ -277,28 +247,44 @@ def __init__(self, bot):
         self, team, tool
     ) -> Union[keybase1.KVDeleteEntryResult, keybase1.KVGetResult, None]:
         expected_revision = 1
-        cached = self.kvstore.get_cached(tool)
-        if cached is not None:
-            expected_revision = cached["revision"] + 1
+        res = await self.lookup(team, tool)
+        expected_revision = res.revision + 1
         res = await self.kvstore.delete(team, self.NAMESPACE, tool, expected_revision)
         return res
 
-    async def update_reservation(
-        self, team, user, tool, day, reserve=True
+    async def reserve(
+        self, team, user, tool, day
     ) -> Union[keybase1.KVPutResult, keybase1.KVGetResult]:
-        # note: if you reserve or unreserve a not-added or deleted tool, it will add the tool
-        info: Dict[str, str] = {}
-        expected_revision = 1
-        cached = self.kvstore.get_cached(tool)
-        if cached is not None:
-            if cached["info"]:
-                info = cached["info"] if type(info) is dict else {}
-            expected_revision = cached["revision"] + 1
-        if reserve:
-            info[day] = user
+        """
+        reserve a tool for a given day if that day is already not reserved.
+        note: if you reserve a not-added or deleted tool, it will add the tool
+        """
+        res = await self.lookup(team, tool)
+        info = json.loads(res.entry_value) if res.entry_value != "" else {}
+        if day in info:
+            return res  # failed to put because day is already reserved.
         else:
-            # unreserve
-            info.pop(day, None)
+            info[day] = user
+        expected_revision = res.revision + 1
+        res = await self.kvstore.put(
+            team, self.NAMESPACE, tool, info, expected_revision
+        )
+        return res
```

Comment:
NAB: maybe here we should return `nil` because (1) you're good to go (and it's generally good practice to separate command from query methods), and (2) to distinguish it from an unsuccessful write

Reply:
or True/False for whether or not the reservation was accepted? (i.e. false if it's not accepted for an expected reason (i.e. someone else has that date))

Reply:
decided to change return type to `Tuple[bool, Union[keybase1.KVGetResult, None]]`